presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewQueryOptimizer.java

```diff
@@ -736,12 +751,64 @@ private RowExpression convertToRowExpression(Expression expression, Scope scope)
                     new Analysis(null, ImmutableMap.of(), false),
                     expression,
                     WarningCollector.NOOP);
-            return SqlToRowExpressionTranslator.translate(
-                    expression,
-                    expressionAnalysis.getExpressionTypes(),
-                    ImmutableMap.of(),
-                    metadata.getFunctionAndTypeManager(),
-                    session);
+        }
+
+        private Expression rewriteExpressionWithCoercions(Expression expression, ExpressionAnalysis analysis)
+        {
+            return ExpressionTreeRewriter.rewriteWith(new CoercionRewriter(analysis), expression, null);
+        }
+
+        private class CoercionRewriter
+                extends ExpressionRewriter<Void>
+        {
+            private final ExpressionAnalysis analysis;
+
+            CoercionRewriter(ExpressionAnalysis analysis)
+            {
+                this.analysis = analysis;
+            }
+
+            @Override
+            public Expression rewriteExpression(Expression node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+            {
+                Expression rewrittenExpression = treeRewriter.defaultRewrite(node, context);
```

Comment:
I am wondering if there is a need to rewrite if coercion information is not available? We should check that first before issuing defaultRewrite?